### PR TITLE
fix: redirect with auto login from 3rd services

### PR DESCRIPTION
**Pull request by phamhieu**
## What kind of change does this PR introduce?

detect for redirect from 3rd party service like vercel, aws ...
- auto trigger github login
- auto redirect to the right place after login

fix vercel integration conflict header key

### CODE REVIEW

File: studio/components/to-be-cleaned/Integration/Vercel.utils.ts [hunk 1]
```diff
@@ -48,7 +48,7 @@ export async function fetchVercelProject({
     url = `${url}?${query}`
   }
   const data = await get(url, {
-    headers: { Authorization: `Bearer ${vercelToken}` },
+    headers: { vercel_authorization: `Bearer ${vercelToken}` },
```
Comment (joshenlim) [inline studio/components/to-be-cleaned/Integration/Vercel.utils.ts:51]: might not matter but wondering if might be good to follow http headers naming convention
https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers

"Vercel-Authorization"

im guessing need to update API too if thats the case

File: studio/pages/index.tsx [hunk 2]
```diff
@@ -103,3 +121,29 @@ const Home: NextPage = () => {
   )
 }
 export default withAuth(observer(Home))
+
+// detect for redirect from 3rd party service like vercel, aws...
+function isRedirectFromThirdPartyService(router: NextRouter) {
+  return router.query.next != undefined || router.query['x-amzn-marketplace-token'] != undefined
```
Comment (joshenlim) [inline studio/pages/index.tsx:127]: `!==` for stricter check

**Review by joshenlim — APPROVED** [review]
some comments but lgtm!

File: studio/components/to-be-cleaned/Integration/Vercel.utils.ts (hunk 1 above)
Comment (phamhieu) [reply studio/components/to-be-cleaned/Integration/Vercel.utils.ts:51]: It's an unofficial header key so i treat it as a normal dictionary key. Should be ok for our use-case

**Pull request merged**